### PR TITLE
Data pipeline update

### PR DESCRIPTION
**Pull request by nimshi89**
### What

Changed the `get_dataset_id` and the `get_submitter_email` now they both take a dictionary that contains the data from the manifest and returns the data needed(I did my best to make sure the changes show what I am attempting to do). I have added a test file `testing.py` to make the testing a bit more easier.

### How to review

Run testing.py and check if the data has been used successfully. 

### Who can review
Anyone

### CODE REVIEW

File: dpypelines/pipeline/shared/utils.py [hunk 1]
```diff
@@ -29,17 +29,19 @@ def get_email_client():
...
     """
 
     # Temporary email address for testing purposes
     # Needs to be updated once we know where the submitter email can be extracted from
+    submitter_email = manifest_dict["fileAuthorEmail"]
 
-    os.environ["TEMPORARY_SUBMITTER_EMAIL"] = "submitter@test.com"
-
-    submitter_email = os.getenv("TEMPORARY_SUBMITTER_EMAIL")
+    if manifest_dict["manifestVersion"] != "1":
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/shared/utils.py]: Value for `manifestVersion` should be an integer not a string

File: dpypelines/pipeline/configuration.py [hunk 2]
```diff
@@ -33,11 +33,11 @@
 }
 
 
-def get_dataset_id(s3_object_name: str) -> str:
+def get_dataset_id(manifest_dict: dict) -> str:
     """
     Placeholder function to be updated once we know where the dataset_id can be extracted from (not necessarily s3_object_name as suggested by argument name)
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/configuration.py]: Update comment now this is no longer a placeholder function

File: dpypelines/pipeline/configuration.py [hunk 3]
```diff
@@ -33,11 +33,11 @@
 }
 
 
-def get_dataset_id(s3_object_name: str) -> str:
+def get_dataset_id(manifest_dict: dict) -> str:
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/configuration.py]: Change function name to `get_source_id`

File: dpypelines/pipeline/configuration.py [hunk 4]
```diff
@@ -33,11 +33,11 @@
...
     """
     Placeholder function to be updated once we know where the dataset_id can be extracted from (not necessarily s3_object_name as suggested by argument name)
     """
-    return "not-specified"
+    return manifest_dict["source_id"]
 
 
 def get_pipeline_config(dataset_id: str) -> dict:
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/configuration.py]: Change argument name to `source_id` and update comment

File: dpypelines/s3_tar_received.py [hunk 5]
```diff
@@ -67,12 +67,23 @@ def start(s3_object_name: str):
     # get_dataset_id() currently empty - returns "not-specified"
```
Comment (SarahJohnsonONS) [inline dpypelines/s3_tar_received.py]: Update comment to reflect changes here

File: dpypelines/s3_tar_received.py [hunk 6]
```diff
@@ -67,12 +67,23 @@ def start(s3_object_name: str):
...
+             "pattern_looked_for": "manifest.json"}
         )
     except Exception as err:
-        logger.error("Failed to retrieve dataset_id", err)
+        logger.error("Failed to to retrieve file: manifest.json", err)
+        de_notifier.failure()
+        raise err
+    
+    try:
+        dataset_id = get_dataset_id(manifest_dict)
```
Comment (SarahJohnsonONS) [inline dpypelines/s3_tar_received.py]: Change `dataset_id` to `source_id` (and in the logger statements on L94 and L99)

File: dpypelines/pipeline/shared/utils.py [hunk 7]
```diff
@@ -29,17 +29,19 @@ def get_email_client():
     return email_client
 
 
-def get_submitter_email() -> str:
+def get_submitter_email(manifest_dict: dict) -> str:
     """
     Placeholder function to be updated once we know where the dataset_id can be extracted from (not necessarily s3_object_name as suggested by argument name)
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/shared/utils.py]: Update comment now this is no longer a placeholder function

**Review by SarahJohnsonONS — COMMENTED** [review]
Code looks good, just a few comments on replacing refs to `dataset_id` (you might want to do a search for other examples and tidy up any other references).

Some of the values in `manifest.json` are needed to populate the parameters for the upload service HTTP request. There's a ticket here to update the HTTP clients which it could be added on to: https://jira.ons.gov.uk/browse/DIS-887. Might be worth checking with Mike, or I can bring it up at refinement tomorrow.

File: dpypelines/s3_tar_received.py (hunk 5 above)
Comment (nimshi89) [reply dpypelines/s3_tar_received.py]: would it be okay to just delete that 2 lines of comment?

File: dpypelines/pipeline/shared/utils.py [hunk 8]
```diff
@@ -4,16 +4,36 @@
 import os
 
 from dpytools.email.ses.client import SesClient
-from dpytools.logging.logger import DpLogger
-from dpytools.utilities.utilities import str_to_bool
 from email_validator import EmailNotValidError, validate_email
 
-logger = DpLogger("data-ingress-pipeline")
+
+def str_to_bool(should_be_bool: str) -> bool:
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/shared/utils.py]: `str_to_bool` should be imported from dpytools.utilities.utilities

**Review by SarahJohnsonONS — COMMENTED** [review]
Code looks good, just one comment. Could you do a find-and-replace for `dataset_id` and `dataset id` as there are still a few references in logging statements etc which could be confusing.

**Review by SarahJohnsonONS — APPROVED** [review]
LGTM

**Pull request merged**